librz/include/rz_core.h

```diff
@@ -826,7 +837,8 @@ RZ_API RzList *rz_core_asm_bwdisassemble(RzCore *core, ut64 addr, int n, int len
 RZ_API RzList *rz_core_asm_back_disassemble_instr(RzCore *core, ut64 addr, int len, ut32 hit_count, ut32 extra_padding);
 RZ_API RzList *rz_core_asm_back_disassemble_byte(RzCore *core, ut64 addr, int len, ut32 hit_count, ut32 extra_padding);
 RZ_API ut32 rz_core_asm_bwdis_len(RzCore *core, int *len, ut64 *start_addr, ut32 l);
-RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, int len, int lines, int invbreak, int nbytes, bool json, PJ *pj, RzAnalysisFunction *pdf);
+RZ_API int rz_core_print_disasm(RZ_NONNULL RzCore *core, ut64 addr, RZ_NONNULL ut8 *buf, int len, int nlines,
+	int invbreak, int cbytes, bool json, RZ_NULLABLE PJ *pj, RZ_NULLABLE RzAnalysisFunction *pdf, RZ_OUT RZ_NULLABLE RzList *out_list);
```

Comment:
Isn't better to use a structure with the config?

Reply:
Agree. Like we made it for strings search function: https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_util/rz_str_search.h#L27